src/app/code/community/Omise/Gateway/Model/Payment/Offsiteinstalment.php

```diff
@@ -0,0 +1,160 @@
+     * @return boolean
+     */
+    public function canUseForCurrency($currencyCode) {
+        return (strtoupper($currencyCode) === 'THB' && strtoupper(Mage::app()->getStore()->getCurrentCurrencyCode()) === 'THB');
+    }
+
+    /**
+     * Instantiate state and set it to state object
+     *
+     * @param string        $payment_action
+     * @param Varien_Object $state_object
+     */
+    public function initialize($payment_action, $state_object) {
+        $payment = $this->getInfoInstance();
+        $order   = $payment->getOrder();
+
+        $invoice = $order->prepareInvoice();
+        $invoice->setIsPaid(false)->register();
+
+        $charge = $this->process($payment, $invoice->getBaseGrandTotal());
+
+        $payment->setCreatedInvoice($invoice)
+                ->setIsTransactionClosed(false)
+                ->setIsTransactionPending(true)
+                ->addTransaction(
+                    Mage_Sales_Model_Order_Payment_Transaction::TYPE_ORDER,
+                    $invoice,
+                    false,
+                    Mage::helper('omise_gateway')->__('Processing an amount of %s via Omise Instalment payment.', $order->getBaseCurrency()->formatTxt($invoice->getBaseGrandTotal()))
+                );
+
+        $order->addRelatedObject($invoice);
+
+        if ($charge->isAwaitPayment()) {
+            $state_object->setState(Mage_Sales_Model_Order::STATE_PENDING_PAYMENT);
+            $state_object->setStatus($order->getConfig()->getStateDefaultStatus(Mage_Sales_Model_Order::STATE_PENDING_PAYMENT));
+            $state_object->setIsNotified(false);
+            return;
+        }
+
+        $this->suspectToBeFailed($payment);
+    }
+
+    /**
+     * @param  Varien_Object $payment
+     * @param  float         $amount
+     *
+     * @return Omise_Gateway_Model_Api_Charge
+     */
+    public function process(Varien_Object $payment, $amount) {
+        $order = $payment->getOrder();
+        $currency = $order->getBaseCurrencyCode();
+
+        return parent::_process(
+            $payment,
+            array(
+                'amount'      => $this->getAmountInSubunits($amount, $currency),
+                'currency'    => $currency,
+                'description' => 'Processing payment with instalments. Magento order ID: ' . $order->getIncrementId(),
+                'source'      => array(
+                    'type' => $payment->getAdditionalInformation('type'),
+                    'installment_terms' => $payment->getAdditionalInformation('terms')
+                ),
+                'return_uri'  => $this->getCallbackUri(),
+                'metadata'    => array(
+                    'order_id' => $order->getIncrementId()
+                )
+            )
+        );
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see app/code/core/Mage/Payment/Model/Method/Abstract.php
+     */
+    public function assignData($data) {
+        parent::assignData($data);
+
+        $type = $data->getData('type');
```

Comment:
@jonrandy should it be `bank_type` instead? To make it consistent with the Internetbanking class (https://github.com/omise/omise-magento/pull/169/files#diff-503a31dfb4ca83262657aef2065a8d6aR114)

Reply:
Thought about that - instalment payments could feasibly be provided by an entity that isn't a bank